### PR TITLE
fix: (Core) Carousel direction defects

### PR DESCRIPTION
**Pull request by olshansky**
#### Please provide a link to the associated issue.
Closes: https://github.com/SAP/fundamental-ngx/issues/4035
Closes: https://github.com/SAP/fundamental-ngx/issues/4036
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/carousel/carousel.component.scss [hunk 1]
```diff
@@ -18,3 +18,9 @@ fd-carousel:focus {
 [dir=rtl] .fd-carousel__button.fd-button > [class*=sap-icon] {
     transform: rotate(0deg);
 }
+
+.fd-carousel__button.fd-button.is-focus,
+.fd-carousel__button.fd-button:focus {
+    top: 50%;
+    left: initial;
+}
```
Comment (KevinOkamoto) [inline libs/core/src/lib/carousel/carousel.component.scss:26]: I think this CSS rule is causing the carousel button to move on focus. This is what I'm seeing when I keyboard navigate to the carousel button:
Screen Shot 2020-12-10 at 8 06 44 AM

**Review by KevinOkamoto — CHANGES_REQUESTED** [review]
See comment.

**Comment by DeepakSap14** [comment]
Buttons are still buggy on chrome browser.

**Comment by olshansky** [comment]
> Buttons are still buggy on chrome browser when placed inside page-indicator.

@DeepakSap14 could you add more details and gifs with examples of issues? Because in my browser all work fine (Chrome 86.0.4240.198)

**Comment by DeepakSap14** [comment]
> > Buttons are still buggy on chrome browser when placed inside page-indicator.
> 
> @DeepakSap14 could you add more details and gifs with examples of issues? Because in my browser all work fine (Chrome 86.0.4240.198)

Yes, it is not reproducible in chrome as well.
May be last time, i went to old deploy link, before new deployment link was available.

**Review by DeepakSap14 — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
There's an RTL service implemented in Core. Can you use it?

**Comment by olshansky** [comment]
> There's an RTL service implemented in Core. Can you use it?

@InnaAtanasova could you clarify what do you mean? 
Carousel component already uses RTL service ([injected here](https://github.com/SAP/fundamental-ngx/blob/d215f587485965e588e9cb473f96e14d7974e905/libs/core/src/lib/carousel/carousel.component.ts#L221) and uses below)

File: libs/core/src/lib/carousel/carousel.component.ts [hunk 2]
```diff
@@ -308,14 +308,14 @@ export class CarouselComponent implements OnInit, AfterContentInit, AfterViewIni
     @HostListener('keydown.arrowright', ['$event'])
     onKeydownArrowRight(event: KeyboardEvent): void {
         event.preventDefault();
-        this.next();
+        this.dir === 'rtl' ? this.previous() : this.next();
```
Comment (stefanoScalzo) [inline libs/core/src/lib/carousel/carousel.component.ts:311]: This should be using the rtl service from core. We have a service that unifies all components to be RTL

**Review by JKMarkowski — COMMENTED** [review]
@olshansky 
Looks good to me, what about creating some `isRTL()` method, that will return boolean based on value from RtlService (it is behaviour subject storing value)?

File: libs/core/src/lib/carousel/carousel.component.ts (hunk 2 above)
Comment (olshansky) [reply libs/core/src/lib/carousel/carousel.component.ts:311]: @stefanoScalzo @InnaAtanasova as I understand you pointed this service – https://github.com/SAP/fundamental-ngx/blob/master/libs/core/src/lib/utils/services/carousel.service.ts
`this.previous()` and `this.next()` methods using this service ([in next](https://github.com/SAP/fundamental-ngx/blob/5de5d4c890d2e53e7f74b4f4682a1cc32e56c1bd/libs/core/src/lib/carousel/carousel.component.ts#L356), [in previous](https://github.com/SAP/fundamental-ngx/blob/5de5d4c890d2e53e7f74b4f4682a1cc32e56c1bd/libs/core/src/lib/carousel/carousel.component.ts#L329)), but it works not in a proper way for RTL direction, and my changes are the fastest solution for this issue

Should I try to fix carousel.service.ts?

**Comment by olshansky** [comment]
> @olshansky
> Looks good to me, what about creating some `isRTL()` method, that will return boolean based on value from RtlService (it is behaviour subject storing value)?

@JKMarkowski added

**Review by InnaAtanasova — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]
LGTM

**Pull request merged**